gems/sorbet-runtime/test/types/types.rb

```diff
@@ -1398,10 +1399,14 @@ def call(x, y)
         assert_match(/Expected type T.class_of\(Opus::Types::Test::TypesTest::Base\), got/, msg)
       end
 
-      it 'works for a mixin' do
```

Comment:
The previous behavior was to treat module singleton classes as not final, despite them actually being final at runtime.

It did this because it used `<=` to compare `self` to `other`, which asks whether **instances** of those types are in an inheritance relationship (e.g., whether an instance of `Mixin1Child` is an instance of `Mixin1`, which is `true)` instead of whether the singleton classes themselves are in an inheritance relationship.